core/src/worker/workflow/mod.rs

```diff
@@ -459,32 +459,34 @@ impl Workflows {
         commands: &mut [Command],
     ) -> Vec<OwnedMeteredSemPermit> {
         let mut reserved = vec![];
-        if let Some(at_handle) = self.activity_tasks_handle.as_ref() {
-            for cmd in commands {
-                if let Some(Attributes::ScheduleActivityTaskCommandAttributes(attrs)) =
-                    cmd.attributes.as_mut()
-                {
-                    // If request_eager_execution was already false, that means lang explicitly
-                    // told us it didn't want to eagerly execute for some reason. So, we only
-                    // ever turn *off* eager execution if a slot is not available or the activity
-                    // is scheduled on a different task queue.
-                    if attrs.request_eager_execution {
-                        let same_task_queue = attrs
-                            .task_queue
+        for cmd in commands {
+            if let Some(Attributes::ScheduleActivityTaskCommandAttributes(attrs)) =
+                cmd.attributes.as_mut()
+            {
+                // If request_eager_execution was already false, that means lang explicitly
+                // told us it didn't want to eagerly execute for some reason. So, we only
+                // ever turn *off* eager execution if a slot is not available or the activity
+                // is scheduled on a different task queue.
+                if attrs.request_eager_execution {
+                    let same_task_queue = attrs
+                        .task_queue
+                        .as_ref()
+                        .map(|q| q.name == self.task_queue)
+                        .unwrap_or_default();
+                    if same_task_queue
+                        && reserved.len() < MAX_EAGER_ACTIVITY_RESERVATIONS_PER_WORKFLOW_TASK
+                    {
+                        if let Some(p) = self
+                            .activity_tasks_handle
```

Comment:
Does `activity_tasks_handle` represent local too? If not, why wouldn't this be `None` if there is no activity worker?

Reply:
Nope. There's a totally separate manager specifically for local activities.

Reply:
Ah, I see now, this didn't _unset_ before because that conditional was outside of the loop.